### PR TITLE
Update xsns_33_ds3231.ino - ESP32-C3

### PR DESCRIPTION
**Pull request by jansenarruda**
Changes to work with ESP32-C3 uint32 (cast to int if processor is ESP32-C3)

## Description:

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [ ] The pull request is done against the latest development branch
  - [ ] Only relevant files were touched
  - [ ] Only one feature/fix was added per PR and the code change compiles without warnings
  - [ ] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with Tasmota core ESP32 V.1.0.7.3
  - [ ] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/xsns_33_ds3231.ino [hunk 1]
```diff
@@ -150,12 +150,22 @@ void DS3231EverySecond(void)
...
     ds3231WriteStatus = true;
   }
+  #else
+  if (!ds3231WriteStatus && Rtc.utc_time > START_VALID_TIME && abs(Rtc.utc_time - ReadFromDS3231()) > 60) {//if time is valid and is drift from RTC in more that 60 second
+    AddLog(LOG_LEVEL_INFO, PSTR("Write Time TO DS3231 from NTP (" D_UTC_TIME ") %s, (" D_DST_TIME ") %s, (" D_STD_TIME ") %s"),
+                GetDateAndTime(DT_UTC).c_str(), GetDateAndTime(DT_DST).c_str(), GetDateAndTime(DT_STD).c_str());
+    SetDS3231Time (Rtc.utc_time); //update the DS3231 time
+    ds3231WriteStatus = true;
+  }
+ #endif
```
[the remark this answers is not in the document]
Comment (jansenarruda) [reply tasmota/xsns_33_ds3231.ino:168]: Hi,
I test only with ESP32 and ESP32-C3, and the error is only with ESP32-C3.

I'll change to int32_t and change only the if statement

[the remark this answers is not in the document]
Comment (jansenarruda) [reply tasmota/xsns_33_ds3231.ino:168]: Another question: I need to made a new pull request? Or I can change this one? (noob question... Very new to github)

[the remark this answers is not in the document]
Comment (jansenarruda) [reply tasmota/xsns_33_ds3231.ino]: Changed as yours comments

**Comment by arendst** [comment]
Thx for noticing.

To make it even easier I will change the code for all processors to:
```
else if (!ds3231WriteStatus && Rtc.utc_time > START_VALID_TIME && abs((int32_t)(Rtc.utc_time - ReadFromDS3231())) > 60) {  // If time is valid and is drift from RTC in more that 60 second
```

**Pull request closed without merging**